app/src/components/ProposalList.tsx

```diff
@@ -16,13 +16,18 @@ type ProposalListState = {
 
 class ProposalList extends React.Component<ProposalListProps, ProposalListState> {
   componentDidMount (): void {
-    this.componentDidUpdate();
+    this.shouldComponentUpdate();
   }
 
-  componentDidUpdate (): void {
+  shouldComponentUpdate (): boolean {
     this.props.drizzle.contracts.Governance.methods.getProposals().call()
-      .then((ids: any) => this.setState({ ids: ids }))
+      .then((ids: any) => {
+        this.setState({ ids: ids });
```

Comment:
I think if `this.setState()` is called within `componentDidUpdate()` or `shouldComponentUpdate()`, it will cause it to recursively update as a change of a component's state will cause it to update and call either of the two methods and so on (try logging a statement in the `componentDidUpdate()` / `shouldComponentUpdate()` methods). This maybe what is causing the bug (putting a `setState()` within these methods without the contract call will cause and error in React of too many recursive calls or something). It seems the reason the UI is responsive at the moment is that its just repeatedly calling getProposals. A better way to update the UI may be to listen to the Propose, Approve, and Reject contract events, but it also doesn't seem like a very clean way to do if we have to listen to 3 different events to update one UI component.

Reply:
Agreed - Not sure what the solution for this one is. Looking for feedback.

Irrespective this isn't the solution